presto-docs/src/main/sphinx/develop/spi-overview.rst

```diff
@@ -30,61 +31,63 @@ single line listing the name of the plugin class:
-plugin to request a service of type ``ConnectorFactory``.
-The connector factory is a simple interface responsible for creating an
-instance of a ``Connector`` object that returns instances of the
-following services:
-
-* ``ConnectorMetadata``
-* ``ConnectorSplitManager``
-* ``ConnectorHandleResolver``
-* ``ConnectorRecordSetProvider``
-
-ConnectorMetadata
-^^^^^^^^^^^^^^^^^
-
-The connector metadata interface has a large number of important
-methods that are responsible for allowing Presto to look at lists of
-schemas, lists of tables, lists of columns, and other metadata about a
-particular data source.
-
-This interface is too big to list in this documentation, but if you
-are interested in seeing strategies for implementing these methods,
-look at the :doc:`example-http` and the Cassandra connector. If
-your underlying data source supports schemas, tables and columns, this
-interface should be straightforward to implement. If you are attempting
-to adapt something that is not a relational database (as the Example HTTP
-connector does), you may need to get creative about how you map your
-data source to Presto's schema, table, and column concepts.
-
-ConnectorSplitManger
-^^^^^^^^^^^^^^^^^^^^
-
-The split manager partitions the data for a table into the individual
-chunks that Presto will distribute to workers for processing.
-For example, the Hive connector lists the files for each Hive
-partition and creates one or more split per file.
-For data sources that don't have partitioned data, a good strategy
-here is to simply return a single split for the entire table. This
-is the strategy employed by the Example HTTP connector.
-
-ConnectorRecordSetProvider
-^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-Given a split and a list of columns, the record set provider is
-responsible for delivering data to the Presto execution engine.
-It creates a ``RecordSet``, which in turn creates a ``RecordCursor``
-that is used by Presto to read the column values for each row.
+to understand the Presto SPI. It contains access methods to retrieve
+various classes that a Plugin can provide. For example, the ``getConnectorFactories()``
+method is a top-level function that Presto calls to retrieve a ``ConnectorFactory`` when Presto
+is ready to create an instance of a connector to back a catalog. There are similar
+methods for ``Type``, ``ParametricType``, ``Function``, ``SystemAccessControl``, and
+``EventListenerFactory`` objects.
+
+Building Plugins via Maven
+--------------------------
+
+Plugins depend on the SPI from Presto:
+
+.. code-block:: xml
+
+    <dependency>
+        <groupId>com.facebook.presto</groupId>
+        <artifactId>presto-spi</artifactId>
+        <scope>provided</scope>
+    </dependency>
+
+The plugin uses the Maven ``provided`` scope because Presto provides
+the classes from the SPI at runtime and thus the plugin should not
+include them in the plugin assembly.
+
+There are a few other dependencies that are provided by Presto such
```

Comment:
This section is out of date now. ``javax.inject`` is no longer in the SPI, and we only provide the Jackson annotations. I'll update.